blocks.md

```diff
@@ -0,0 +1,111 @@
+
+The block name (meaning the [`name` parameter to `registerBlockType()`](https://developer.wordpress.org/block-editor/developers/block-api/block-registration/#block-name) and [`name` in `block.json`](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#name)) must be unique to your block. As with titles, please avoid names containing trademarks or names you do not legally represent.
+
+The namespace prefix to the block name should reflect either the plugin author, or the plugin slug. For example:
+
+`name: "my-rainbow-block-plugin/rainbow"`, or
+`name: "john-doe/rainbow"`, or
+`name: "pluginco/rainbow"`.
+
+The namespace must not be a reserved one such as `core`.
+
+<h4>4. Block Plugins must include a <code>block.json</code> file.</h4>
+
+The Block Registration RFC outlines the format of a `block.json` file: https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md
+
+Block Plugins must include a valid `block.json` file. In addition to the requirements specified in the RFC, the `block.json` must include the following attributes:
+* name
+* title
+* At least one of: script, editorScript
+* At least one of: style, editorStyle
+
+<h4>5. Block Plugins must not require payment or a paid account to function.</h4>
+
+This includes payment for premium features. You may use the donation link feature.
+
+Block plugins that do require a paid service or include upselling and premium features are still permitted in the main WordPress Plugin Directory, subject to its guidelines.
+
+<h4>6. Block Plugins should work seamlessly.</h4>
+
+Block Plugins are intended to work seamlessly and instantly when installed from the editor. That means they should not encumber the user with additional steps or prerequisites such as installing another plugin or theme, signing up for an account, or logging in or manually connecting to an external service.
+
+Block Plugins must not require another plugin in order to function.
+
+They may use an external/cloud API where necessary, provided it can be done seamlessly without requiring a login or activation key.
+
+They should not rely on an external API or cloud service for functions that could be performed locally.
+
+<h4>7. Server-side code should be kept to a minimum.</h4>
+
+Since Block Plugins are WordPress plugins, they necessarily contain PHP code for metadata and initialisation. As much as possible, they should avoid including additional PHP code or server-side libraries. The WordPress REST API should be the preferred interface to WordPress, rather than custom server-side code.
+
+We recognise that there are limits to the REST API, and situations where server-side PHP is the only performant way to implement a feature. In those situations, PHP code may be included. Please ensure that such code is clearly written, used as little as possible, and well documented.
+
+<h4>8. Must not include advertisements or promote other blocks, plugins, or themes.</h4>
```

Comment:
There was a conversation yesterday in #meta that I wanted to get clarity on in line with the guidlines.

https://wordpress.slack.com/archives/C02QB8GMM/p1594850305004600

Using my "Jetpack Business Hours" example. That block is currently part of Jetpack, but it is relatively trivial for it to live in a single/ad-hoc block plugin.

With the assumption that we handle a case where someone has both installed, would it be allowed under these guidelines for it to live in both Jetpack and in an ad-hoc plugin.

For the sake of the conversation, I'm not concerned with it potentially being extra work for the developer to keep in sync or whether it is practical :), just would that ad hoc plugin be allowed while the identical block is still within a block collection/other plugin.

Reply:
Currently allowed, but **strongly** recommended against, with the expectation that eventually you would remove it from Jetpack. And it would be disallowed for new plugins going the other way.

We don't want some developer to have 30 plugins, where 29 are ad-hoc-blocks and 1 is a collection of all their blocks. That's just gaming the system, making more work for themselves (you don't care, fine, but other people really need to remember that they're not Jetpack/large company with many hands making light work), and making it a right bloody mess for your users who are trying to find a plugin.

Basically ... if you're going to decouple your blocks, fully decouple them. We'd give allowances for people who (like you) had blocks included in a larger plugin but wanted to break them out later (since you pre-dated the advent of this sub-directory) but the end goal needs to be an either/or, not a both.